crates/engine/tree/src/tree/mod.rs

```diff
@@ -1902,7 +1905,15 @@ where
         head: &SealedBlock<N::Block>,
     ) -> ProviderResult<Option<PayloadStatus>> {
         // check if the check hash was previously marked as invalid
-        let Some(header) = self.state.invalid_headers.get(&check) else { return Ok(None) };
+        let Some(header) = self.state.invalid_headers.get(&check) else {
+            if self.state.invalid_headers.get(head.hash_ref()).is_some() {
+                // the head hash has already been marked as invalid previously
+                let status = self.prepare_invalid_response(head.parent_hash())?;
```

Comment:
note: here we would use `LinksToRejectedPayload`, do we mb update the docs for this variant to say "links to or is a rejected invalid payload"

Reply:
oh right we should construct the status manually here